src/AccessionTables/components/LocationSelection/LocationSelection.js

```diff
@@ -3,33 +3,54 @@ import PropTypes from 'prop-types';
 import { escapeRegExp } from 'lodash';
 import { useIntl } from 'react-intl';
 
+import { OptionSegment } from '@folio/stripes/components';
 import { Selection } from '@folio/stripes-acq-components';
 
 import { Location } from '../Location';
 
 // todo: this is forked from stripes-smart-components, got to figure out the way to merge
 
-// todo: does not highlights the search criterion in the results
-const filter = (value, data) => {
-  return data.filter(o => new RegExp(escapeRegExp(value), 'i').test(o.label.props?.location.name));
+const filter = (searchString, dataOptions) => {
+  const regexp = new RegExp(escapeRegExp(searchString), 'i');
+  const match = location => Location.textFields.some(field => regexp.test(location[field]));
+
+  return dataOptions.filter(match);
 };
 
 
-export const LocationSelection = ({ locations, placeholder, ...rest }) => {
+const Formatter = ({ option, searchTerm }) => {
   const { formatMessage } = useIntl();
 
-  const finalPlaceholder = placeholder || formatMessage({ id: 'stripes-smart-components.ls.locationPlaceholder' });
+  if (option.value === '') return formatMessage({ id: 'ui-remote-storage.notSet' });
```

Comment:
Maybe we could use this form? `if (!option.value)` or we need compare it with empty string only?

Reply:
thanks, fixed